epitator/geoname_annotator.py

```diff
@@ -211,9 +213,17 @@ def __init__(self, geoname, spans_to_nes, span_to_tokens):
         d['num_spans'] = len(geoname.spans)
         d['max_span_length'] = max([
             len(span.text) for span in geoname.spans])
-        d['cannonical_name_used'] = 1 if any([
-            span.text == geoname.name for span in geoname.spans
-        ]) else 0
+        def cannonical_name_match(span, geoname):
+            if hasattr(span, "iterate_leaf_base_spans"):
+                span_text = next(span.iterate_leaf_base_spans()).text
+            else:
+                span_text = span.text
+            span_in_name = span_text in geoname.name or span_text in geoname.asciiname
+            return (float(len(span_text)) if span_in_name else 0) / len(geoname.name)
```

Comment:
Should we have a hasattr on the name for geoname here, to make sure we don't divide by zero?

Reply:
name is a required attribute for all geonames.